core/service.go

```diff
@@ -41,15 +46,16 @@ var _ services.Service = &Service{}
 // BABE session, and network service. It deals with the validation of transactions
 // and blocks by calling their respective validation functions in the runtime.
 type Service struct {
-	blockState   BlockState
-	storageState StorageState
-	rt           *runtime.Runtime
-	bs           *babe.Session
-	keys         []crypto.Keypair
-	blkRec       <-chan types.Block     // receive blocks from BABE session
-	msgRec       <-chan network.Message // receive messages from network service
-	epochDone    <-chan struct{}        // receive from this channel when BABE epoch changes
-	msgSend      chan<- network.Message // send messages to network service
+	blockState        BlockState
+	storageState      StorageState
+	rt                *runtime.Runtime
+	bs                *babe.Session
+	keys              []crypto.Keypair
+	blkRec            <-chan types.Block     // receive blocks from BABE session
+	msgRec            <-chan network.Message // receive messages from network service
+	epochDone         <-chan struct{}        // receive from this channel when BABE epoch changes
+	msgSend           chan<- network.Message // send messages to network service
+	requestedBlockIDs map[uint64]bool        // track requested block id messages
```

Comment:
Should we import and store this in network state? Would it be useful within core?

Reply:
cc @noot This would be a reason to keep network state and I am now imaging other possible scenarios where we might want to use network state and have network state separate from other state.

Reply:
do you mean the requestedBlockIDs ?

Reply:
Yes, I am asking about storing the requestedBlockIDs

Reply:
yeah it could be useful to have in the network state as opposed to core

Reply:
ok